Fix issue when creating dataTypes

CristianGimenez opened the pull request:
This PR will be marked as "Draft" because it depends on a [PR for mdm-resources](https://github.com/MauroDataMapper/mdm-resources/pull/104).

The main cause of the issue was that `new-data-type-inline.component` was using an anonymous type with the wrong variable names, and `default-profile-editor-modal.component` was using those variable names without treating them. All this is allowed because the `DataType`definition allows `[key : string]: value` as a variable.

There is a number of changes in this PR:
- Removing unused validations and some small fixes that I spotted while working in the different components.
- Removing anonymous types to use `DataType`.
- Adding some comments on the code to describe better what needs to happen or why we do certain transformations.
- When creating a new `DataType` the label for the new `dataType `is mandatory, so we should show the fields from the start.

Both creating a new `dataType` while editing an existing `dataElement `and when creating a brand new `dataElement `should work fine now.


closes #318

CristianGimenez committed:
gh-318: small bugfix and house cleaning

CristianGimenez committed:
gh318: new datatype while editing default profile works

Along this changes, there is eslint changes and removing duplicated validation code

CristianGimenez committed:
gh-318: adding new elements now working

CristianGimenez committed:
gh-318: prettifier changes

CristianGimenez committed:
gh-318: removing dataType option to have unknown variables

abwilson23 commented on src/app/dataElement/data-element/data-element.component.ts:
```diff
@@ -221,7 +215,20 @@ export class DataElementComponent
         resource.maxMultiplicity = item.maxMultiplicity;
       } else if (item.controlType === ProfileControlTypes.dataType) {
-        resource.dataType = item.value as DataTypeReference;
+        const castedValue = item.value as DataType;
```
Not a necessary change but given we've identified that `item.controlType` is `datatype` a more descriptive name than `castedValue` might be `dataTypeValue`.

CristianGimenez replied:
Makes sense, changing.

abwilson23 commented on src/app/data-type/data-type.component.ts:
```diff
@@ -129,8 +132,6 @@ export class DataTypeComponent extends BaseComponent
         this.dataType.classifiers = this.dataType.classifiers || [];
         this.loadingData = false;
-        this.showExtraTabs =
-          !this.sharedService.isLoggedIn() || !this.dataType.editable;
```
Just a note that you can remove the injected SharedService from the the constructor as this was the only use of it.

CristianGimenez replied:
Done

abwilson23 reviewed (dismissed):
Couple of small comments, but overall code changes appear sensible and concise. The functional testing went well, changing data types and adding new ones has the described and desired behaviour. 

It appears there are some jenkins tests to sort out still before it can be merged.

CristianGimenez committed:
gh-318:  Pr comments

CristianGimenez commented:
@abwilson23 I've addressed your comments in the latest commit. Could you resolve the comments if you are happy? Whatever Jenkins was unhappy about seems to be ok now.

abwilson23 reviewed (approved):
Approved ✅

jamesrwelch commented:
@abwilson23 @CristianGimenez I'm happy to merge this now, if you're both happy?

CristianGimenez commented:
> @abwilson23 @CristianGimenez I'm happy to merge this now, if you're both happy?

Yes please, thanks!

jamesrwelch commented:
Fantastic work both of you 👍
